viral/ensemble_reactivation.py

```
@@ -0,0 +1,617 @@
+
+def compute_ICA_components(ssp_vectors: np.ndarray) -> np.ndarray:
+    """Subsequently, the ICA components of these smoothed run
+    activity estimates were calculated across time using the fastICA algorithm. Only
+    the subset of components found to be significant under the Marcenko-Pasteur
+    distribution were considered ICA ensembles and included in the ICA ensemble
+    matrix w, with rows corresponding to PCs and columns corresponding to
+    significant ICA components."""
+    ica_transfomer = FastICA()
+    # expected input: (n_samples, n_features)
+    # independent components (frames, components)
+    independent_components = ica_transfomer.fit_transform(ssp_vectors.T)
```

Ruff (0.8.2)

Local variable `independent_components` is assigned to but never used

Remove assignment to unused variable `independent_components`

(F841)